### PR TITLE
[Flink-K8s-V2] Port the new module codes to Streampark-flink-kubernetes-v2 

### LINKED ISSUE #2881: [Feature] [Flink-K8s-V2] Port the new module codes to Streampark-flink-kubernetes-v2
It is a subtask of #2879 which includes the following content:
- Add new module `streampark-flink/streampark-flink-kubernetes-v2`
    - Port basic utility code and replace some of the code with existing facilities in StreamPark:
        - [util](https://github.com/Al-assad/streampark-flink-kubernetes-v2/tree/master/src/main/scala/org/apache/streampark/flink/kubernetes/util) package
        - [K8sTool](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/K8sTools.scala)
    - Port the embedded HTTP file service located at [hfs](https://github.com/Al-assad/streampark-flink-kubernetes-v2/tree/master/src/main/scala/org/apache/streampark/flink/kubernetes/hfs);
    - Port  FlinkK8sObserver located in the [observer](https://github.com/Al-assad/streampark-flink-kubernetes-v2/tree/master/src/main/scala/org/apache/streampark/flink/kubernetes/observer) package;
    - Port FlinkK8sOperator located in the [operator](https://github.com/Al-assad/streampark-flink-kubernetes-v2/tree/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator) package;
    - Adapt the new module to StreamPark's infrastructure:
        - Integrate `org.apache.streampark.console.core.runner.EnvInitializer` with the startup logic of [FileServer](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/hfs/FileServer.scala)
        - Add the following configuration from flink-k8s-v2 into StreamPark's configuration system :
            - [hdf/package](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/hfs/package.scala)
            - [observer/package](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/observer/package.scala)
            - [operator/package](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/package.scala)

Design document: https://www.craft.me/s/LrTTzYHqSOLYqY

### PR DESCRIPTION
**Pull request by Al-assad**
## What changes were proposed in this pull request

Issue Number: close #2881 

This is the foundational work for the streampark flink-k8s-v2 series tasks #2879. The main task of this PR is to provide an independent core module that will not affect any existing operational mechanisms in streampark.

## Verifying this change

This change added tests and can be verified as follows:
 - org/apache/streampark/flink/kubernetes/v2/example/UsingEmbeddedFileServer.scala
 - org/apache/streampark/flink/kubernetes/v2/example/UsingObserver.scala
 - org/apache/streampark/flink/kubernetes/v2/example/UsingOperator.scala
 - org/apache/streampark/flink/kubernetes/v2/FlinkMemorySizeParserSpec.scala

## Does this pull request potentially affect one of the following parts
 - Dependencies:
   - https://mvnrepository.com/artifact/com.lihaoyi/pprint_2.12/0.8.1 MIT
   - https://mvnrepository.com/artifact/com.lihaoyi/os-lib_2.12/0.8.1 MIT
   - https://mvnrepository.com/artifact/com.lihaoyi/upickle_2.12/0.8.1 MIT
   - https://mvnrepository.com/artifact/io.fabric8/kubernetes-client/6.8.0 Apache-2.0
   - https://mvnrepository.com/artifact/dev.zio/zio_2.12/2.0.15 Apache-2.0
   - https://mvnrepository.com/artifact/dev.zio/zio-streams_2.12/2.0.15 Apache-2.0
   - https://mvnrepository.com/artifact/dev.zio/zio-concurrent_2.12/2.0.15 Apache-2.0
   - https://mvnrepository.com/artifact/dev.zio/zio-http_2.12/3.0.0-RC2 Apache-2.0

  <br/>

### CODE REVIEW

**Comment by Al-assad** [comment]
@wolfboys @GOODBOY008 PTAL

File: streampark-common/pom.xml [hunk 1]
```diff
@@ -128,6 +128,32 @@
             <version>${streampark.shaded.version}</version>
         </dependency>
 
+        <!-- ZIO -->
+        <dependency>
```
Comment (wolfboys) [inline streampark-common/pom.xml:132]: "common" is a public module that many other modules depend on. It is not a good idea to add ZIO dependency in the common module.

**Comment by wolfboys** [comment]
Overall, it looks good, but there are still some minor issues. We can merge this pull request first and improve it later.

**Review by wolfboys — APPROVED** [review]

File: streampark-common/pom.xml (hunk 1 above)
Comment (Al-assad) [reply streampark-common/pom.xml:132]: The interfaces exposed by streampark-flink-kubernetes are all zio or zstream structures, and external calls must refer to zio. I'll be writing more zio code in other modules later on.ZIO is a basic, pure-scala, zero-dependency library, so placing it in comon doesn't seem like a problem.

**Pull request merged**